components/camel-infinispan/src/main/java/org/apache/camel/component/infinispan/policy/InfinispanRoutePolicy.java

```diff
@@ -0,0 +1,516 @@
+        }
+
+        if (!leader.get() && shouldStopConsumer) {
+            stopConsumer(route);
+        }
+    }
+
+    @Override
+    public synchronized void onStop(Route route) {
+        try {
+            stopService();
+        } catch (Exception e) {
+            throw new RuntimeCamelException(e);
+        }
+
+        suspendedRoutes.remove(route);
+    }
+
+    @Override
+    public synchronized void onSuspend(Route route) {
+        try {
+            stopService();
+        } catch (Exception e) {
+            throw new RuntimeCamelException(e);
+        }
+
+        suspendedRoutes.remove(route);
+    }
+
+    @Override
+    protected void doStart() throws Exception {
+        // validate
+        StringHelper.notEmpty(lockMapName, "lockMapName", this);
+        StringHelper.notEmpty(lockKey, "lockKey", this);
+        StringHelper.notEmpty(lockValue, "lockValue", this);
+        ObjectHelper.notNull(camelContext, "camelContext", this);
+
+        if (this.lockValue == null) {
+            this.lockValue = camelContext.getUuidGenerator().generateUuid();
+        }
+
+        this.manager.start();
+        this.executorService = getCamelContext().getExecutorServiceManager().newSingleThreadScheduledExecutor(this, "InfinispanRoutePolicy");
+
+        BasicCache<String, String> cache = manager.getCache(lockMapName);
+        if (manager.isCacheContainerEmbedded()) {
+            this.service = new EmbeddedCacheService(InfinispanUtil.asEmbedded(cache));
+        } else {
+            this.service = new RemoteCacheService(InfinispanUtil.asRemote(cache));
+        }
+
+        super.doStart();
+    }
+
+    @Override
+    protected void doStop() throws Exception {
+        if (future != null) {
+            future.cancel(true);
+            future = null;
+        }
+
+        if (this.service != null) {
+            this.service.stop();
+        }
+
+        getCamelContext().getExecutorServiceManager().shutdownGraceful(executorService);
+
+        leader.set(false);
+        manager.stop();
+
+        super.doStop();
+    }
+
+    private void startService() throws Exception {
+        if (service == null) {
+            throw new IllegalStateException("A RoutePolicyService should be configured");
+        }
+
+        service.start();
+    }
+
+    private void stopService() throws Exception {
+        leader.set(false);
+
+        if (this.service != null) {
+            this.service.stop();
+        }
+    }
+
+    // *************************************************************************
+    //
+    // *************************************************************************
+
+    protected void setLeader(boolean isLeader) {
+        if (isLeader && leader.compareAndSet(false, isLeader)) {
+            LOGGER.info("Leadership taken (map={}, key={}, val={})", lockMapName, lockKey, lockValue);
+
+            startAllStoppedConsumers();
+        } else {
+            if (!leader.getAndSet(isLeader) && isLeader) {
```

Comment:
can someone review this condition? I don't think the log is printed correctly. 
If I understand correctly, it should be ```leader.getAndSet(isLeader) && !isLeader```. ```leader``` was ```true``` and it was set to ```false```.